lib/file.coffee

```diff
@@ -67,3 +67,6 @@ class File
 
   isPathEqual: (pathToCompare) ->
     @path is pathToCompare or @realPath is pathToCompare
+
+  # Return a snapshot of the instance's filesystem properties for API consumption
+  getMetadata: -> {@symlink, @realPath, @status, @stats}
```

Comment:
`*cough*` UH, well... at the moment, it's returning a reference to the calling [`TabView`](https://github.com/atom/tabs/blob/master/lib/tab-view.coffee#L215) element, which I'll obviously tend to once we're done with this PR

Access to the tab's `File` instance is available through `@item.buffer.file`, but that's an instance of [one of these](https://atom.io/docs/api/v1.8.0/File), which doesn't have the cool extras of filesystem stats the Tree-View's `File` class has.